src/utils/drag-and-drop-utils.js

```diff
@@ -68,6 +68,9 @@ function canDrop(dropTargetProps, monitor, isHover = false) {
 
     const targetDepth = getTargetDepth(dropTargetProps, monitor);
     const draggedNode = monitor.getItem().node;
+    const parentPath = dropTargetProps.path.slice(0, -1);
```

Comment:
Not sure this is a reliable way to retrieve the parent node. Most notably, it doesn't work while hovering, so the preview still shows up and the drop is refused only when the user releases the element. I think it would be a better UX to prevent the preview from showing up already while dragging (just like the `maxDepth` param already does).

I'd like to have this feature too so I'm willing to help on implementing this. I already looked a bit at the code but I'm not sure how to do this yet. Most notably, the `path` doesn't reflect the exact final position  of the node while dragging and we probably need to use the `targetDepth` also. I'll have a deeper look at how to do this next week so hopefully I can submit some improvements to your PR.

@fritz-c, would you have any advice on how to reliably retrieve the expected parent node while dragging ? Just looking at the row above is not enough since it also depends on the target depth. Maybe we would have to traverse the rows before until we find the expected parent.